Azure blob integration tests broken by context propagation

Eg https://pub.rclone.org/integration-tests/2019-06-28-050155/

See https://pub.rclone.org/integration-tests/current for the most recent

![image](https://user-images.githubusercontent.com/536803/60346073-4e654980-99b2-11e9-9d5f-33f23aa0cc93.png)

These all appear to have something to do with renaming directories...

According to git bisect this was the commit which broke it f78cd1e0434ee597fb8937ba3cda0d5fd58becfe

This added context into the interface and passed the context on to the AzureBlob SDK which doesn't look as if it should have broken anything.

The error is this

```
$ ( cd fs/operations && go test -v -remote TestAzureBlob: -run TestDirMove -verbose )
=== RUN   TestDirMove
2019/06/28 14:53:16 DEBUG : A1/B2: Making directory
2019/06/28 14:53:16 DEBUG : A1/B1/C3: Making directory
2019/06/28 14:53:16 INFO  : A1/B1/three: Copied (server side copy)
2019/06/28 14:53:16 INFO  : A1/one: Copied (server side copy)
2019/06/28 14:53:16 INFO  : A1/two: Copied (server side copy)
2019/06/28 14:53:16 INFO  : A1/B1/C1/four: Copied (server side copy)
2019/06/28 14:53:16 INFO  : A1/B1/three: Deleted
2019/06/28 14:53:16 INFO  : A1/one: Deleted
2019/06/28 14:53:16 INFO  : A1/two: Deleted
2019/06/28 14:53:16 INFO  : A1/B1/C1/four: Deleted
2019/06/28 14:53:16 INFO  : A1/B1/C2/five: Copied (server side copy)
2019/06/28 14:53:16 INFO  : A1/B1/C2/five: Deleted
--- FAIL: TestDirMove (0.44s)
    run.go:172: Remote "Azure container rclone-test-jiganin7cujuvex3zahebab9", Local "Local file system at /tmp/rclone881171044", Modify Window "1ns"
    fstest.go:260: Filtering empty directory "A1/B2"
    fstest.go:260: Filtering empty directory "A1/B1/C3"
    require.go:794: 
        	Error Trace:	operations_test.go:1168
        	Error:      	Received unexpected error:
        	            	-> github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline.NewError, /home/ncw/go/src/github.com/ncw/rclone/vendor/github.com/Azure/azure-pipeline-go/pipeline/error.go:154
        	            	HTTP request failed
        	            	
        	            	Get https://rclone.blob.core.windows.net/rclone-test-jiganin7cujuvex3zahebab9?comp=list&delimiter=&include=metadata&maxresults=1&prefix=A1%2FB1%2FC2%2F&restype=container&timeout=31536001: context canceled
        	            	
        	            	RenameDir rmdir
        	            	github.com/ncw/rclone/fs/operations.DirMove
        	            		/home/ncw/go/src/github.com/ncw/rclone/fs/operations/operations.go:1802
        	            	github.com/ncw/rclone/fs/operations_test.TestDirMove
        	            		/home/ncw/go/src/github.com/ncw/rclone/fs/operations/operations_test.go:1168
        	            	testing.tRunner
        	            		/opt/go/go1.12/src/testing/testing.go:865
        	            	runtime.goexit
        	            		/opt/go/go1.12/src/runtime/asm_amd64.s:1337
        	Test:       	TestDirMove
FAIL
2019/06/28 14:53:16 DEBUG : Azure container rclone-test-jiganin7cujuvex3zahebab9: Purge remote
exit status 1
FAIL	github.com/ncw/rclone/fs/operations	0.475s
```

Making this very small change (stopping the context propagation into the Azure SDK) fixes the problem

```diff
diff --git a/backend/azureblob/azureblob.go b/backend/azureblob/azureblob.go
index 92c6893ec..0c54057fb 100644
--- a/backend/azureblob/azureblob.go
+++ b/backend/azureblob/azureblob.go
@@ -520,6 +520,7 @@ type listFn func(remote string, object *azblob.BlobItem, isDirectory bool) error
 //
 // dir is the starting directory, "" for root
 func (f *Fs) list(ctx context.Context, dir string, recurse bool, maxResults uint, fn listFn) error {
+	ctx = context.Background()
 	f.containerOKMu.Lock()
 	deleted := f.containerDeleted
 	f.containerOKMu.Unlock()
```

Any ideas @sandeepkru or @ajankovic ?

--- a/fs/operations/multithread.go
+++ b/fs/operations/multithread.go
@@ -122,9 +122,9 @@ func multiThreadCopy(ctx context.Context, f fs.Fs, remote string, src fs.Object,
 		return nil, errors.New("multi-thread copy: can't copy zero sized file")
 	}
 
-	g, ctx := errgroup.WithContext(context.Background())
+	g, gCtx := errgroup.WithContext(ctx)
 	mc := &multiThreadCopyState{
-		ctx:     ctx,
+		ctx:     gCtx,
 		size:    src.Size(),
 		src:     src,
 		streams: streams,
@@ -136,7 +136,7 @@ func multiThreadCopy(ctx context.Context, f fs.Fs, remote string, src fs.Object,
 	defer fs.CheckClose(mc.acc, &err)
 
 	// create write file handle
-	mc.wc, err = openWriterAt(ctx, remote, mc.size)
+	mc.wc, err = openWriterAt(gCtx, remote, mc.size)
 	if err != nil {
 		return nil, errors.Wrap(err, "multpart copy: failed to open destination")
 	}
@@ -146,7 +146,7 @@ func multiThreadCopy(ctx context.Context, f fs.Fs, remote string, src fs.Object,
 	for stream := 0; stream < mc.streams; stream++ {
 		stream := stream
 		g.Go(func() (err error) {
-			return mc.copyStream(ctx, stream)
+			return mc.copyStream(gCtx, stream)
 		})
 	}
 	err = g.Wait()

--- a/fs/operations/operations.go
+++ b/fs/operations/operations.go
@@ -1762,18 +1762,18 @@ func DirMove(ctx context.Context, f fs.Fs, srcRemote, dstRemote string) (err err
 		newPath string
 	}
 	renames := make(chan rename, fs.Config.Transfers)
-	g, ctx := errgroup.WithContext(context.Background())
+	g, gCtx := errgroup.WithContext(context.Background())
 	for i := 0; i < fs.Config.Transfers; i++ {
 		g.Go(func() error {
 			for job := range renames {
-				dstOverwritten, _ := f.NewObject(ctx, job.newPath)
-				_, err := Move(ctx, f, dstOverwritten, job.newPath, job.o)
+				dstOverwritten, _ := f.NewObject(gCtx, job.newPath)
+				_, err := Move(gCtx, f, dstOverwritten, job.newPath, job.o)
 				if err != nil {
 					return err
 				}
 				select {
-				case <-ctx.Done():
-					return ctx.Err()
+				case <-gCtx.Done():
+					return gCtx.Err()
 				default:
 				}
 

